src/groupeddataframe/groupeddataframe.jl

```diff
@@ -323,6 +323,14 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function Base.to_index(gd::GroupedDataFrame, key::Tuple{Vararg{<:Pair}})
```

Comment:
I think it is better to be more restrictive with the signature here:
```suggestion
function Base.to_index(gd::GroupedDataFrame, key::Union{Tuple{Vararg{Pair{Symbol}}}), Tuple{Vararg{Pair{<:AbstractString}}})}
```